### PR TITLE
New API: Response.peekBody

### PR DESCRIPTION
**Pull request by swankjesse**
Closes: https://github.com/square/okhttp/issues/1740

### CODE REVIEW

File: okhttp-tests/src/test/java/okhttp3/ResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/ResponseTest.java]: !

**Comment by swankjesse** [comment]
Test flake unrelated to current PR. Sigh!
`ConnectionReuseTest.staleConnectionNotReusedForNonIdempotentRequest`

File: okhttp-tests/src/test/java/okhttp3/ResponseTest.java (hunk 1 above)
Comment (JakeWharton) [inline okhttp-tests/src/test/java/okhttp3/ResponseTest.java]: year of ~~the linux desktop~~ okhttp3 and retrofit2!

File: okhttp-tests/src/test/java/okhttp3/ResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  /**
+   * Returns a new response body that refuses to be read once it has been closed. This is true of
+   * most {@link BufferedSource} instances, but not of {@link Buffer}.
+   */
+  private ResponseBody responseBody(String content) {
+    final Buffer data = new Buffer().writeUtf8(content);
+
+    Source source = new Source() {
```
Comment (JakeWharton) [inline okhttp-tests/src/test/java/okhttp3/ResponseTest.java]: I think `Okio.buffer(new ForwardingSource(data) {})` does the trick here.

Comment (JakeWharton) [inline okhttp-tests/src/test/java/okhttp3/ResponseTest.java]: Actually looks like `Okio.buffer((Source) data)` might as well.

**Pull request merged**